geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/TransformSimpleFeature.scala

```diff
@@ -0,0 +1,118 @@
+  override def getID: String = underlying.getID
+
+  override def getUserData: jMap[AnyRef, AnyRef] = underlying.getUserData
+
+  override def getType: SimpleFeatureType = transformSchema
+  override def getFeatureType: SimpleFeatureType = transformSchema
+  override def getName: Name = transformSchema.getName
+
+  override def getAttribute(name: Name): AnyRef = getAttribute(name.getLocalPart)
+  override def getAttribute(name: String): Object = {
+    val index = transformSchema.indexOf(name)
+    if (index == -1) null else getAttribute(index)
+  }
+
+  override def getDefaultGeometry: AnyRef = getAttribute(geomIndex)
+  override def getAttributeCount: Int = transformSchema.getAttributeCount
+
+  override def getBounds: BoundingBox = getDefaultGeometry match {
+    case g: Geometry => new ReferencedEnvelope(g.getEnvelopeInternal, transformSchema.getCoordinateReferenceSystem)
+    case _           => new ReferencedEnvelope(transformSchema.getCoordinateReferenceSystem)
+  }
+
+  override def getAttributes: jList[AnyRef] = {
+    val attributes = new java.util.ArrayList[AnyRef](transformSchema.getAttributeCount)
+    var i = 0
+    while (i < transformSchema.getAttributeCount) {
+      attributes.add(getAttribute(i))
+      i += 1
+    }
+    attributes
+  }
+
+  override def getDefaultGeometryProperty = throw new NotImplementedError
+  override def getProperties: jCollection[Property] = throw new NotImplementedError
+  override def getProperties(name: Name) = throw new NotImplementedError
+  override def getProperties(name: String) = throw new NotImplementedError
+  override def getProperty(name: Name) = throw new NotImplementedError
+  override def getProperty(name: String) = throw new NotImplementedError
+  override def getValue = throw new NotImplementedError
+  override def getDescriptor = throw new NotImplementedError
+
+  override def setAttribute(name: Name, value: Object) = throw new NotImplementedError
+  override def setAttribute(name: String, value: Object) = throw new NotImplementedError
+  override def setAttribute(index: Int, value: Object) = throw new NotImplementedError
+  override def setAttributes(vals: jList[Object]) = throw new NotImplementedError
+  override def setAttributes(vals: Array[Object]) = throw new NotImplementedError
+  override def setDefaultGeometry(geo: Object) = throw new NotImplementedError
+  override def setDefaultGeometryProperty(geoAttr: GeometryAttribute) = throw new NotImplementedError
+  override def setValue(newValue: Object) = throw new NotImplementedError
+  override def setValue(values: jCollection[Property]) = throw new NotImplementedError
+
+  override def isNillable = true
+  override def validate() = throw new NotImplementedError
+
+  override def toString = s"TransformSimpleFeature:$getID"
+}
+
+object TransformSimpleFeature {
+
+  import scala.collection.JavaConversions._
+
+  def apply(sft: SimpleFeatureType, transformSchema: SimpleFeatureType, transforms: String): TransformSimpleFeature = {
```

Comment:
Why would it have to move to index-api?

Meh, if I can figure out a way to sort out the call I'm thinking off, I'll add it to this class.